warehouse/static/js/warehouse/utils/bind-filter-keys.js

```diff
@@ -0,0 +1,49 @@
+/**
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+export default () => {
+  // Force tab key to cycle within the filter panel if the filter panel is active.
+  const keys = [];
+  document.addEventListener("keydown", event => {
+    const filterPanel = document.querySelector(".filter-panel");
+    keys[event.keyCode] = event.type === "keydown";
+
+    if (filterPanel && keys[9]) {
+      // `firstItem` assumes that the close button is the first tabbable element.
+      const firstItem = filterPanel.querySelector(".filter-panel__close");
+
+      const getLastItem = () => filterPanel.querySelector(
+        ".accordion:last-child").classList.contains("accordion--closed")
+        ? filterPanel.querySelector(".accordion:last-child > .accordion__link")
+        : filterPanel.querySelector(".accordion:last-child li:last-child > input");
```

Comment:
@nlhkabu These selectors could use your 👀: they work for me, but I'm not sure if they're as efficient/semantic as they could be.

Reply:
Maybe a better way to do this would be to add a class `.filter-panel__last-item` to both the link and input?  my concern here is that if we add more content to the panel (e.g. anything that is not an accordion) we will mess up the script.

At least if we can keep the selectors specific to the `filter-panel` component, we can document this class in the SCSS file for future reference.  Wdyt?

Reply:
> Maybe a better way to do this would be to add a class `.filter-panel__last-item` to both the link and input? my concern here is that if we add more content to the panel (e.g. anything that is not an accordion) we will mess up the script.

Yep, I agree, but I'm not actually sure about the best way to do that here -- we generate the classifier tree with a recursive macro, and there might not be a clean way to determine the last child `input`.



> At least if we can keep the selectors specific to the `filter-panel` component, we can document this class in the SCSS file for future reference. Wdyt?

👍